include/GoogleSync/GoogleSync.php

```diff
@@ -1002,71 +1156,69 @@ protected function setLastSync($event_local, $gEventId = null)
             return false;
         }
 
         if (!$this->setUsersGoogleCalendar()) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to setup Google Calendar Id');
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to setup Google Calendar Id');
             return false;
         }
 
         $meetings = $this->getUserMeetings();
         if (!isset($meetings)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to get Users Meetings');
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to get Users Meetings');
             return false;
         }
 
         // First, we look for SuiteCRM meetings that are not on Google
         foreach ($meetings as $meeting) {
-            $gevent = null; // Sanity Check. At the begining of each loop, $gevent is NULL.
             $gevent = $this->getGoogleEventById($meeting->gsync_id);
             $dowhat = $this->pushPullSkip($meeting, $gevent);
 
             switch ($dowhat) {
                 case "push":
-                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pushing Meeting: ' . $meeting->name);
+                    $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pushing Meeting: ' . $meeting->name);
                     $this->pushEvent($meeting, $gevent);
                     break;
                 case "pull":
-                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pulling Meeting: ' . $meeting->name);
+                    $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pulling Meeting: ' . $meeting->name);
                     $this->pullEvent($gevent, $meeting);
                     break;
                 case "skip":
-                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Skipping Meeting: ' . $meeting->name);
+                    $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Skipping Meeting: ' . $meeting->name);
                     break;
                 case "push_delete":
-                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Event: ' . $meeting->name);
+                    $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Event: ' . $meeting->name);
                     $this->delEvent($gevent, $meeting->id);
                     break;
                 case "pull_delete":
-                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Meeting: ' . $meeting->name);
+                    $this->logger->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Meeting: ' . $meeting->name);
                     $this->delMeeting($meeting);
```

Comment:
delMeeting could return false as failure. Please handle the error.

Reply:
return is now passed to caller